### ISSUE TITLE
Game Icon by TitleID

### ISSUE THREAD

**Issue by ohhsodead** [aldostools/Resources#13]
Is there any way to get the icon from the titleid?

I have seen you have the covers dumped and it's a nice alternative so thanks for that.

But are there any endpoints in the PSN API that could fetch the PS3 games data or the icon?

I know there is something similar for PS4;

`https://store.playstation.com/store/api/chihiro/00_09_000/titlecontainer/BE/nl/999/CUSA01163_00`
`https://store.playstation.com/store/api/chihiro/00_09_000/titlecontainer/BE/nl/999/CUSA01163_00/image`

I'm sure you are aware of the above, if not I hope you find some use from it!

**Comment by ohhsodead** [comment]
Ah, never mind. I think I found an alternative

**Comment by aldostools** [comment]
Here is an example:
`https://store.playstation.com/store/api/chihiro/00_09_000/container/GB/en/19/EP0001-BLES00909_00-AC2DLC4000000001`

**Comment by ohhsodead** [comment]
> Here is an example: `https://store.playstation.com/store/api/chihiro/00_09_000/container/GB/en/19/EP0001-BLES00909_00-AC2DLC4000000001`

Oh I didn't know it still held data for PS3 games. Thank you!

**Comment by ohhsodead** [comment]
> Here is an example: `https://store.playstation.com/store/api/chihiro/00_09_000/container/GB/en/19/EP0001-BLES00909_00-AC2DLC4000000001`

Are you aware of any links or formats that can only use the TitleId rather than ContentId?

**Comment by aldostools** [comment]
No. AFAIK all the PS3 contents in the PSN Store should have a single content ID.

A single TitleID could have multiple contents associated.

**Comment by ohhsodead** [comment]
@aldostools

How could one find the ContentId?

I have made a function to scrape the webman extension webpage to get the TitleId and the Game Title, could this be used to get the contentId in any way? Are there any files on the console that its stored on? I already tried the PARAM.SFO file but this ContentId is always empty/blank.

**Comment by aldostools** [comment]
The content id is found only in digital content (DLC, updates, PSN games, etc.). Retail games (ISO) don't have a content id.

The content id can be found in the header of the EBOOT.BIN or SELF files. It's near the offset 0x430 of the file.

The PKG files also include a content id.

**Comment by ohhsodead** [comment]
> The content id is found only in digital content (DLC, updates, PSN games, etc.). Retail games (ISO) don't have a content id.
> 
> The content id can be found in the header of the EBOOT.BIN or SELF files. It's near the offset 0x430 of the file.
> 
> The PKG files also include a content id.

When you say "updates", is that this folder; `/dev_hdd0/game/{REGION}` - if so, where exactly could it be?

Sorry for being a pain, just I don't have my PS3 with me anymore so I lack being able to check myself. I appreciate your help. Thank you.

**Comment by aldostools** [comment]
I mean the game update packages, PSN games, homebrews and DLC. They are installed in /dev_hdd0/game

 The eboots and self files are signed as npDRM (free or licensed). Therefore they have a content id. However the eboots and self files found in blu-ray discs or ISO of PS3 games are signed differently without a content id.
